grin/src/Reducer/Pure.hs

```diff
@@ -21,94 +25,113 @@ prettyDebug :: Pretty a => a -> String
 prettyDebug = show . plain . pretty
 
 -- models computer memory
-data StoreMap
+data StoreMap v
   = StoreMap
-  { storeMap  :: IntMap RTVal
+  { storeMap  :: IntMap (RTVal v)
   , storeSize :: !Int
   }
 
 emptyStore = StoreMap mempty 0
 
+type ValueConstraints v = (Eq v, Show v, Pretty v)
+data EvalPlugin v = EvalPlugin
+  { evalPluginPrimOp  :: Name -> [Val] -> [RTVal v] -> IO (RTVal v)
```

Comment:
Why do we need the both the second and third arguments?

Reply:
It is the type signature for the
```evalPrimOp :: MonadIO m => Name -> [Val] -> [RTVal Lit] -> m (RTVal Lit)```
in the PrimOps module. I don't want to do an extensive refactor of that one.